app/controllers/admin/attachments_controller.rb

```diff
@@ -14,7 +15,9 @@ def order
     redirect_to attachable_attachments_path(attachable), notice: "Attachments re-ordered"
   end
 
-  def new; end
+  def new
+    render(preview_design_system_user? ? "new" : "new_legacy")
```

Comment:
If the fail is unsuccessful it will rerender the edit view here https://github.com/alphagov/whitehall/blob/87af8214f730585c3273613ee4a31f5ad26a06f6/app/controllers/admin/attachments_controller.rb#L26-L28

We will also need this logic 
```
 render(preview_design_system_user? ? "new" : "new_legacy")```

there so the correct page is rerendered